### DISCUSSION TITLE
Need help to post a message to a thread

### DISCUSSION THREAD

**Discussion by BlutVampir** [discord-php/DiscordPHP#1084] [Q&A] [answered]
I really struggle with a lot of things

I want to post a message to a thread.
I have the thread_id which can get via: $channel->threads ...

I tried to set: 
$channel = $discord->getChannel($thread_id); which is not working and delivers a NULL.

I tried to:
$message = MessageBuilder::new()->setContent("Debug: test")->setReplyTo($thread_id);

which is also not woking. How can I get a Message posted to the thread?

Additionally I want afterwards to get all messages of the thread to:

$channel->getMessageHistory(['limit' => 50])->done(function (Collection $messages) {
	print_r($messages);
});

this code is not woking, equalless if in a cannel or thread..

Can a Function

$thread = $discord->getChannel($thread_id)->getThread(thread_id); be implemented?

Or how can I manage these things?

**Comment by SQKo** [comment, marked as answer]
`$discord->getChannel()` can only lookup for a cached channel, not threads.
`setReplyTo()` can be only used to another message object, not a thread id.

You will have to get the channel by id first then get the thread by the id.

**Reply by civilCornball** [reply]
So, like this?

```
<?php

//Assuming $channel_id and $thread_id are correctly defined previously

// Fetch the channel first (you're already doing this)
$channel = $interaction->guild->channels->get('id', $channelId);

// Now, attempt to fetch the thread directly from the channel
$channel->threads->fetch($threadId)->then(
    function ($thread) {
        // Success: $thread is your desired thread object
        echo "Thread found: ", $thread->name, PHP_EOL;
        // You can now interact with your $thread object here
    },
    function ($e) {
        // Error handling: Could not find the thread or an error occurred
        echo "Error fetching thread: ", $e->getMessage(), PHP_EOL;
    }
);
```

**Reply by SQKo** [reply]
Yes
